collector/snyk/src/server.rs

```diff
@@ -93,65 +90,135 @@ pub async fn collect_packages(
+                            }
+                        } else {
+                            collected_snyk_errors.push(format!("purl error: {}", purl));
+                        }
+                    }
+                    Err(err) => collected_snyk_errors.push(err.to_string()),
+                }
+            }
+            Err(_) => {
+                collected_snyk_errors.push(format!("unable to rewrite {original_purl}"));
+            }
         }
+    }
 
-        if !vulns.is_empty() {
-            state
-                .guac_client
-                .intrinsic()
-                .ingest_package(&PackageUrl::from_str(purl).map_err(|_| Error::Internal)?.into())
-                .await
-                .map_err(|_| Error::Guac)?;
-
-            for vuln in &vulns {
-                state
-                    .guac_client
-                    .intrinsic()
-                    .ingest_vulnerability(&VulnerabilityInputSpec {
-                        r#type: "snyk".to_string(),
-                        vulnerability_id: vuln.id.clone(),
-                    })
-                    .await
-                    .map_err(|_| Error::Guac)?;
-
-                state
-                    .guac_client
-                    .intrinsic()
-                    .ingest_certify_vuln(
-                        &PackageUrl::from_str(purl).map_err(|_| Error::Internal)?.into(),
-                        &VulnerabilityInputSpec {
-                            r#type: "snyk".to_string(),
-                            vulnerability_id: vuln.id.clone(),
-                        },
-                        &ScanMetadataInput {
-                            db_uri: "https://api.snyk.io/".to_string(),
-                            db_version: "1.0".to_string(),
-                            scanner_uri: "https://trustification.io/".to_string(),
-                            scanner_version: "1.0".to_string(),
-                            time_scanned: Default::default(),
-                            origin: "snyk".to_string(),
-                            collector: "snyk".to_string(),
-                        },
-                    )
-                    .await
-                    .map_err(|_| Error::Guac)?;
-
-                state.v11y_client.ingest_vulnerability(vuln).await.ok();
+    // now stuff all the expanded issue->vulns into v11y proper.
+    if !vulns.is_empty() {
+        for vuln in &vulns {
+            if let Err(err) = state.v11y_client.ingest_vulnerability(vuln).await {
+                collected_v11y_errors.push(err.to_string())
             }
         }
     }
 
-    Ok(HttpResponse::Ok().finish())
+    let gathered = CollectPackagesResponse {
+        purls: gathered_purls,
+        errors: collected_snyk_errors
+            .iter()
+            .map(|err| err.to_string())
+            .chain(collected_guac_errors.iter().map(|err| err.to_string()))
+            .chain(collected_v11y_errors.iter().map(|err| err.to_string()))
```

Comment:
I'm curious why all the map calls are needed since it appears every `push` call param is already a String?

Reply:
Thanks, adjusted!